common/app/views/fragments/atoms/snippets/timeline.scala.html

```diff
@@ -1,6 +1,18 @@
 @(timeline: model.content.TimelineAtom, isAmp: Boolean)(implicit request: RequestHeader)
 
 @import org.joda.time.format.DateTimeFormat
+@import com.gu.contentatom.thrift.atom.timeline.TimelineItem
+
+@renderFormattedDate(item: TimelineItem, format: String) = @{
+   item.toDate match {
+       case Some(date) => {
+            val beginRange = DateTimeFormat.forPattern(format).print(item.date)
+            val endRange = DateTimeFormat.forPattern(format).print(date)
+            s"$beginRange&ndash;$endRange"
+       }
+       case None => DateTimeFormat.forPattern(format).print(item.date)
+   }
+}
```

Comment:
Can we move any logic out of tempalte so it can be unit tested for instance? what about a `toFormattedDate` function in `TimelineAtome`.

Reply:
Are you suggesting creating a companion object `TimelineAtom` in `Atoms.scala` which is called as a standalone function? Or something different that would involve preprocessing of data?

Reply:
I was thinking about a method in either the TimelineAtom class or a companion object.